pkg/api/start.go

```diff
@@ -356,10 +362,17 @@ func getRetryInterval() time.Duration {
 	return 5 * time.Second // Default to 5 seconds if not set
 }
 
-// isRunningInCluster checks if the application is running inside cluster
-func isRunningInCluster() bool {
+// isRunningInCluster checks if the application is running in cluster
+func isRunningInCluster() (bool, error) {
 	_, err := rest.InClusterConfig()
-	return err == nil
+
+	if err == rest.ErrNotInCluster {
+		return false, nil
+	} else if err != nil {
+		return true, err
```

Comment:
If there's a an err, recommened we respond with a negative

```suggestion
		return false, err
```

Reply:
i was thinking about that, but i think we still want to affirm if it's in cluster, more particularly for when it's called by the health check handler. if there's an error but it's in cluster then we still want to not send an actual cluster check that requires the kubecontext from local.